clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/AWSServerGroupNameResolver.groovy

```diff
@@ -45,6 +52,55 @@ class AWSServerGroupNameResolver extends AbstractServerGroupNameResolver {
 
   @Override
   List<AbstractServerGroupNameResolver.TakenSlot> getTakenSlots(String clusterName) {
-    return asgService.getTakenSlots(clusterName)
+    def cluster = awsCluster(clusterProviders, accountName, clusterName)
+    if (!cluster) {
+      return []
+    }
+
+    def serverGroupsInRegion = cluster.serverGroups.findAll { it.region == region }
+    return serverGroupsInRegion.collect {
+      new AbstractServerGroupNameResolver.TakenSlot(
+        serverGroupName: it.name,
+        sequence       : Names.parseName(it.name).sequence,
+        createdTime    : new Date(it.createdTime)
+      )
+    }
+  }
+
+  @Override
+  String resolveNextServerGroupName(String application, String stack, String details, Boolean ignoreSequence) {
+    def originalNextServerGroupName = super.resolveNextServerGroupName(application, stack, details, ignoreSequence)
+    def nextServerGroupName = originalNextServerGroupName
+    def clusterName = combineAppStackDetail(application, stack, details)
+
+    if (nextServerGroupName) {
+      while (true) {
```

Comment:
thinking being that the time it takes to do 1000 distinct getAutoScalingGroup calls is probably going to blow things out and we should have just failed after 5? attempts